### PR TITLE
Restart the dev container on code changes

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Run Nodemon rather than Node directly. Changes to the code now cause the process to restart, rather than having to restart the container itself.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Nodemon appears to add a _lot_ of dependencies.

File: package.json [hunk 1]
```diff
@@ -6,12 +6,13 @@
...
     "@types/node": "^12.12.6",
+    "nodemon": "^1.19.4",
     "ts-node": "^8.4.1",
     "typescript": "^3.7.2"
   },
   "scripts": {
     "build": "tsc",
     "start": "node ./build/index.js",
-    "start:dev": "node --require ts-node/register ./src/index.ts"
+    "start:dev": "nodemon --exec 'node --require ts-node/register ./src/index.ts' --ext .js,.json,.ts"
```
Comment (thewilkybarkid) [inline package.json:23]: I remember having problems before when trying `nodemon` instead a Docker container, but seems ok at the moment?

**Comment by giorgiosironi** [comment]
Running:
- `make build`
- `make start`
- `docker logs -f docker_app_1`
- edit `src/app.ts`

works:
```
> @ start:dev /app
> nodemon --exec 'node --require ts-node/register ./src/index.ts' --ext .js,.json,.ts

[nodemon] 1.19.4
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,json,ts
[nodemon] starting `node --require ts-node/register ./src/index.ts`
[nodemon] restarting due to changes...
```

File: package-lock.json [hunk 2]
```diff
@@ -135,23 +141,248 @@
...
+      "version": "4.0.0",
+      "resolved": "https://registry.npmjs.org/arr-diff/-/arr-diff-4.0.0.tgz",
+      "integrity": "sha1-1kYQdP6/7HHn4VI1dhoyml3HxSA=",
+      "dev": true
+    },
+    "arr-flatten": {
+      "version": "1.1.0",
+      "resolved": "https://registry.npmjs.org/arr-flatten/-/arr-flatten-1.1.0.tgz",
+      "integrity": "sha512-L3hKV5R/p5o81R7O02IGnwpDmkp6E982XhtbuwSe3O4qOtMMMtodicASA1Cny2U+aCXcNpml+m4dPsvsJ3jatg==",
+      "dev": true
```
Comment (giorgiosironi) [inline package-lock.json:384]: at least they are all `"dev": true`

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**